TodoListService/Startup.cs

```diff
@@ -49,13 +52,25 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddAuthentication(sharedOptions =>
-            {
-                sharedOptions.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
-            })
-            .AddAzureAdBearer(options => Configuration.Bind("AzureAd", options));
+			// This is required to be instantiated before the OpenIdConnectOptions starts getting configured.
+			// By default, the claims mapping will map claim names in the old format to accommodate older SAML applications.
+			// 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' instead of 'roles'
+			// This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
+			JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+
+			services.AddAuthentication(sharedOptions =>
+			{
+				sharedOptions.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
+			})
+			.AddAzureAdBearer(options => Configuration.Bind("AzureAd", options));
+
+			services.Configure<OpenIdConnectOptions>(AzureADDefaults.OpenIdScheme, options =>
```

Comment:
This is a Web API? how can there be OpenIdConnectOptions ? I'm confused?
You would need to Configure the JwtBearerTokenOptions?

Reply:
@jmprieur if I dont set this configuration, I cant use the word `roles` as claim type. I would need to use `http://schemas.microsoft.com/ws/2008/06/identity/claims/role` instead. 
If you want me to remove it, it is fine, but we will need to accept the usage of the long string for role claim type.

Reply:
@TiagoBrenck : I'm saying that this shoud probably be:

```CSharp
services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
```

rather than 

```CSharp
services.Configure<OpenIdConnectOptions>(AzureADDefaults.OpenIdScheme, options =>
```
?